app/src/androidTest/java/ch/epfl/sdp/blindwar/tutorial/DemoActivityTest.kt

```diff
@@ -21,44 +28,112 @@ class DemoActivityTest {
+            "Geralt didnt calm her with the Sign he jumped from the saddle and threw the reins over the horses head " +
+            "He no longer had his old sword in its lizard skin sheath on his back; its place was filled with a shining " +
+            "beautiful weapon with a cruciform and slender well-weighted hilt ending in a spherical pommel made of white metal" +
+            "This time the gate didnt open for him It was already open, just as he had left it" +
+            " He heard singing He didnt understand the words he couldnt even identify the language " +
+            "He didnt need to the witcher felt and understood the very nature the essence of this quiet " +
+            "piercing singing which flowed through the veins in a wave of nauseous overpowering menace"
+
+    private val round = Tutorial
+        .gameInstance
+        .gameConfig
+        .parameter
+        .round
+
+    private fun checkLayoutVisibility(visibility: Visibility) {
+        onView(withId(R.id.guessButton)).check(matches(withEffectiveVisibility(visibility)))
+        onView(withId(R.id.guessEditText)).check(matches(withEffectiveVisibility(visibility)))
+        onView(withId(R.id.scoreTextView)).check(matches(withEffectiveVisibility(visibility)))
+        //onView(withId(R.id.startButton)).check(matches(withEffectiveVisibility(visibility)))
+    }
+
+    private fun makeCorrectGuess() {
+        var correctMetadata = SongMetaData("", "", "")
+        testRule.scenario.onActivity {
+            correctMetadata = it.game.currentMetadata()!!
+        }
+
+        //Log.d(TAG, correctMetadata.toString())
+        onView(withId(R.id.guessEditText))
+            .perform(clearText(), typeText(correctMetadata.title), closeSoftKeyboard())
+        onView(withId(R.id.guessButton)).perform(click())
+    }
+
+    private fun makeBadGuess() {
+        onView(withId(R.id.guessEditText))
+            .perform(clearText(), typeText("THIS IS NOT CORRECT"), closeSoftKeyboard())
+        onView(withId(R.id.guessButton)).perform(click())
+    }
+
+    private fun makeGoodGuessGetBack() {
+        makeCorrectGuess()
+        testRule.scenario.onActivity {
+            it.onBackPressed()
+        }
     }
 
     @Test
-    fun isGuessClickable() {
-        onView(withId(R.id.guessButton)).check(matches(isClickable()))
+    fun songSummaryDisplayedAfterCorrectGuess() {
+        makeCorrectGuess()
+        onView(withId(R.id.song_summary_fragment)).check(matches(isDisplayed()))
     }
 
+
+    /**
+    /** 30 seconds to guess **/
     @Test
-    fun guessResetAfterClick() {
+    fun timeOutTest() {
         onView(withId(R.id.guessEditText))
-            .perform(clearText(), typeText("THIS IS NOT CORRECT"))
-        onView(withId(R.id.guessButton)).perform(closeSoftKeyboard()).perform(click())
-        onView(withId(R.id.guessEditText)).check(matches(withText("")))
+            .perform(clearText(), typeText(longString))

```

Comment:
There are a lot of commented sections like this that should be deleted when not needed, it makes the code a bit messy.